### PR TITLE
Updates to documentation

### PR DESCRIPTION
**Pull request by jdebacker**
This PR makes the following updates to the Tax-Calculator documentation:
1. Removes the "Upgrading to Tax-Calculator 2.0" section of the Jupyter Book (per Issue #2523).
2. Updates the documentation of `iitax` per Issue #2514.
3. Adds Google Colab launch to Jupyter Book, per Issue #2429.
4. Adds `behresp` to the environment.yml` file so that recipes can build from environment, per Issue #2426
5. Updates labels for tables per Issue #2396.
6. Drops use of `disutils.core` in the `setup.py` file per suggestion in Issue #2428.
7. Update recipes to ensure they work on Google Colab and use reform files on the web (see Issue #2429).

### CODE REVIEW

File: taxcalc/records_variables.json [hunk 1]
```diff
@@ -720,8 +720,8 @@
     "iitax": {
       "type": "float",
       "desc": "Total federal individual income tax liability; appears as INCTAX variable in tc CLI minimal output",
-      "form": {"2013-2013": "1040 line 61 minus line 56 minus line 60a",
-               "2014-2016": "1040 line 63 minus line 57 minus line 62a"}
+      "form": {"2013-2013": "1040 line 61 minus lines 56, 60a, 64a, 65, and 66",
```
Comment (jdebacker) [inline taxcalc/records_variables.json:723]: @donboyd5 How do you feel about this update to the documentation of the computation of `iitax` from form 1040?

File: taxcalc/utils.py [hunk 2]
```diff
@@ -58,24 +58,24 @@
                       'expanded_income',
                       'aftertax_income']
 
-DIST_TABLE_LABELS = ['Returns',
+DIST_TABLE_LABELS = ['Number of Returns',
```
Comment (jdebacker) [inline taxcalc/utils.py:61]: @MaxGhenis See changes to labels on distribution and difference tables for those columns that are counts.  Changed things to "Number of Returns" in both.

cc @andersonfrailey

File: docs/recipes/recipe06.ipynb [hunk 3]
```diff
@@ -195,7 +196,7 @@
     "policy2 = tc.Policy()  # parametric reform, reformC.json, eliminates EITC\n",
     "\n",
     "# TODO: Move to the web so this can be done standalone.\n",
-    "policy2.implement_reform(tc.Policy.read_json_reform('_static/reformC.json'))\n",
+    "policy2.implement_reform(tc.Policy.read_json_reform('github://PSLmodels:Tax-Calculator@master/docs/recipes/_static/reformC.json'))\n",
```
Comment (jdebacker) [inline docs/recipes/recipe06.ipynb:199]: To facilitate running on Google Colab, I changed the recipes to file reforms at URLs (need to use this URL format due to issue with `fsspec` dependency of paramtools having issues with other github URL formats at the moment, see [ParamTools Issue #128](https://github.com/PSLmodels/ParamTools/issues/128)).

This works fine, only issue would be if one wanted to change recipes and run locally before those are committed to the branches referenced.

Let me know if you think this is a problem or if you see other issues with this approach.

cc @MaxGhenis @MattHJensen

File: taxcalc/utils.py (hunk 2 above)
Comment (jdebacker) [reply taxcalc/utils.py:61]: @MaxGhenis and @andersonfrailey Do these labels look fine with you?

File: taxcalc/records_variables.json (hunk 1 above)
Comment (jdebacker) [reply taxcalc/records_variables.json:723]: @donboyd5 Let me know if you think these updates are satisfactory.  Thanks!

**Comment by jdebacker** [comment]
@MattHJensen I've taken off the "WIP" and marked this as ready for review.  There are two questions outstanding to people who opened issues that are addressed here.  But those are just confirmation - I believe I addressed their issues.

The biggest change to note here -- for Google Colab functionality, recipes now pull from reform json files via URL rather than relative file paths.  That does assume a user of those notebooks is connected to the internet.

File: taxcalc/records_variables.json (hunk 1 above)
Comment (donboyd5) [reply taxcalc/records_variables.json:723]: @jdebacker thanks for this, it looks great, and sorry for the slow response.

File: taxcalc/utils.py (hunk 2 above)
Comment (MaxGhenis) [reply taxcalc/utils.py:61]: LGTM, or `# returns` if the table gets too wide with this

**Review by MaxGhenis — COMMENTED** [review]
.

**Comment by MattHJensen** [comment]
Looks great, thanks for all of these cleanups and improvements, @jdebacker and thanks for review @MaxGhenis.

**Pull request merged**